applications/minotari_ledger_wallet/wallet/src/utils.rs

```diff
@@ -228,52 +248,26 @@ pub fn get_key_from_canonical_bytes<T: ByteArray>(bytes: &[u8]) -> Result<T, App
     }
 }
 
+/// Get the domain separated alpha key hasher
 pub fn alpha_hasher(
-    alpha: Zeroizing<RistrettoSecretKey>,
-    blinding_factor: Zeroizing<RistrettoSecretKey>,
-) -> Result<Zeroizing<RistrettoSecretKey>, AppSW> {
-    let hasher = DomainSeparatedHasher::<Blake2b<U64>, KeyManagerTransactionsHashDomain>::new_with_label("script key");
-    let hasher = hasher.chain(blinding_factor.as_bytes()).finalize();
-    let private_key = get_key_from_uniform_bytes(hasher.as_ref())?;
-
-    Ok(Zeroizing::new(private_key.deref() + alpha.deref()))
-}
-
-pub fn derive_from_bip32_key(
-    u64_account: u64,
-    u64_index: u64,
-    u64_key_type: KeyType,
-) -> Result<Zeroizing<RistrettoSecretKey>, AppSW> {
-    let account = u64_to_string(u64_account);
-    let index = u64_to_string(u64_index);
-    let key_type = u64_to_string(u64_key_type.as_byte() as u64);
-
-    let mut bip32_path = "m/44'/".to_string();
-    bip32_path.push_str(&BIP32_COIN_TYPE.to_string());
-    bip32_path.push_str(&"'/");
-    bip32_path.push_str(&account);
-    bip32_path.push_str(&"'/0/");
-    bip32_path.push_str(&index);
-    bip32_path.push_str(&"'/");
-    bip32_path.push_str(&key_type);
-    let path: [u32; 6] = make_bip32_path(bip32_path.as_bytes());
+    alpha: RistrettoSecretKey,
+    blinding_factor: RistrettoSecretKey,
+) -> Result<RistrettoSecretKey, AppSW> {
+    let mut raw_key_hashed = Zeroizing::new([0u8; 64]);
+    DomainSeparatedHasher::<Blake2b<U64>, KeyManagerTransactionsHashDomain>::new_with_label("script key")
+        .chain(blinding_factor.as_bytes())
+        .finalize_into(raw_key_hashed.as_mut().into());
+    let private_key = get_key_from_uniform_bytes(&raw_key_hashed)?;
 
-    match get_raw_key(&path) {
-        Ok(val) => get_key_from_uniform_bytes(&val.as_ref()),
-        Err(e) => {
-            SingleMessage::new(&format!("Key error {:?}", e)).show_and_wait();
-            return Err(AppSW::KeyDeriveFail);
-        },
-    }
+    Ok(private_key + alpha)
 }
 
-pub fn get_random_nonce() -> Result<Zeroizing<RistrettoSecretKey>, AppSW> {
+/// Get a uniform random nonce
+pub fn get_random_nonce() -> Result<RistrettoSecretKey, AppSW> {
     let mut raw_bytes = [0u8; 64];
     LedgerRng.fill_bytes(&mut raw_bytes);
     if raw_bytes == [0u8; 64] {
         return Err(AppSW::RandomNonceFail);
     }
-    Ok(Zeroizing::new(
-        RistrettoSecretKey::from_uniform_bytes(&raw_bytes).expect("will not fail"),
-    ))
+    Ok(RistrettoSecretKey::from_uniform_bytes(&raw_bytes).expect("will not fail"))
```

Comment:
We shouldn't use the `expect` here. We need to return a proper `AppSW` error code if it fails.

Reply:
Good catch